metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/DropWizardMetricFactory.java

```diff
@@ -62,6 +64,13 @@ public TimeMetric timer(String name) {
         return Flux.from(publisher).doOnComplete(timer::stopAndPublish);
     }
 
+    @Override
+    public <T> Publisher<T> runPublishingTimerMetricLogP99(String name, Publisher<T> publisher) {
+        TimeMetric timer = timer(name);
```

Comment:
Because the timer calculation starts when creating the object (check the constructor). If you create it in the `doOnComplete`, you will miss your metric, because you time it after the operation is done, while you want to time the operation itself